core/src/main/java/org/apache/calcite/adapter/jdbc/JdbcRules.java

```diff
@@ -766,7 +766,7 @@ public RelNode convert(Sort sort, boolean convertInputTraits) {
 
       final RelNode input;
       if (convertInputTraits) {
-        input = convert(sort.getInput(), traitSet);
+        input = convert(sort.getInput(), out);
```

Comment:
Here `traitSet` is the set of output traits which includes collation (i.e. it tells that the result set is sorted). But it is not necessarily the case for input node. So for conversion need to replace only the convention trait (`out`) for the input leaving other potential traits intact.